lib/ansible/modules/monitoring/zabbix_template.py

```diff
@@ -0,0 +1,296 @@
+        required: false
+    clear_templates:
+        description:
+            - List of templates cleared from the template.
+            - see templates_clear in https://www.zabbix.com/documentation/3.0/manual/api/reference/template/update
+        required: false
+    macros:
+        description:
+            - List of templates macro
+        required: false
+    state:
+        description:
+            - state present create/update template, absent delete template
+        required: false
+        choices: [ present, absent]
+        default: "present"
+    timeout:
+        description:
+            - The timeout of API request (seconds).
+        required: false
+        default: 10
+'''
+
+EXAMPLES = '''
+---
+- name: create templates
+  local_action:
+    module: zabbix_template
+    server_url: http://127.0.0.1
+    login_user: username
+    login_password: password
+    template_name: ExampleHost
+    template_groups:
+      - Role
+      - Role2
+    link_templates:
+      - Example template1
+      - Example template2
+    clear_templates:
+      - Example template3
+      - Example template4
+    macros:
+      - macro: '{$EXAMPLE_MACRO1}'
+        value: 30000
+      - macro: '{$EXAMPLE_MACRO2}'
+        value: 3
+      - macro: '{$EXAMPLE_MACRO3}'
+        value: 'Example'
+    state: present
+'''
+
+RETURN = '''
+#defaults
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from zabbix_api import ZabbixAPI
+
+    HAS_ZABBIX_API = True
+except ImportError:
+    HAS_ZABBIX_API = False
+
+
+class Template(object):
+    def __init__(self, module, zbx):
+        self._module = module
+        self._zapi = zbx
+
+    # check if host group exists
+    def check_host_group_exist(self, group_names):
+        for group_name in group_names:
+            result = self._zapi.hostgroup.get({'filter': {'name': group_name}})
+            if not result:
+                self._module.fail_json(msg="Hostgroup not found: %s" %
+                                       group_name)
+        return True
+
+    # get group ids by group names
+    def get_group_ids_by_group_names(self, group_names):
+        group_ids = []
+        if self.check_host_group_exist(group_names):
+            group_list = self._zapi.hostgroup.get(
+                {'output': 'extend',
+                 'filter': {'name': group_names}})
+            for group in group_list:
+                group_id = group['groupid']
+                group_ids.append({'groupid': group_id})
+        return group_ids
+
+    def get_template_ids(self, template_list):
+        template_ids = []
+        if template_list is None or len(template_list) == 0:
+            return template_ids
+        for template in template_list:
+            template_list = self._zapi.template.get(
+                {'output': 'extend',
+                 'filter': {'host': template}})
+            if len(template_list) < 1:
+                return None
```

Comment:
Do you really want to return here or should this not just `continue` with the next iteration?